homeassistant/components/shelly/switch.py

```diff
@@ -60,90 +82,27 @@ async def async_setup_entry(
 ) -> None:
     """Set up switches for device."""
     if get_device_entry_gen(config_entry) in RPC_GENERATIONS:
```

Comment:
I think you could write this like

```python
    entry_data = get_entry_data(hass)[config_entry.entry_id]

    if get_device_entry_gen(config_entry) in RPC_GENERATIONS:
        if (
            (rpc_coordinator := entry_data.rpc)
            and rpc_coordinator.model == MODEL_WALL_DISPLAY
            and not rpc_coordinator.device.shelly.get("relay_in_thermostat", False)
        ):
            # Wall Display relay is not used as the thermostat actuator,
            # we need to remove a climate entity
            unique_id = f"{rpc_coordinator.mac}-thermostat:0"
            async_remove_shelly_entity(hass, "climate", unique_id)

        return async_setup_entry_rpc(
            hass, config_entry, async_add_entities, RPC_SWITCHES, RpcRelaySwitch
        )

    if (block_coordinator := entry_data.block) and block_coordinator.model is MODEL_GAS:
        return async_setup_entry_attribute_entities(
            hass, config_entry, async_add_entities, GAS_VALVE_SWITCH, BlockValveSwitch
        )

    return async_setup_entry_attribute_entities(
        hass, config_entry, async_add_entities, SWITCHES, BlockRelaySwitch
    )
```

It was a bit nicer to have separate functions for RPC and BLOCK devices though since it meant we didn't have to branch at different levels of indent